infra-deployment-scripts/create-dns-zones.sh

```diff
@@ -1,30 +1,42 @@
 #!/bin/bash
 
-# Check if the GOOGLE_CLOUD_PROJECT environment variable is set
+# Check to see if the environment variable GOOGLE_CLOUD_PROJECT has been set
 if [ -z "${GOOGLE_CLOUD_PROJECT}" ]; then
-    echo "Environment variable GOOGLE_CLOUD_PROJECT is not set."
+    echo "Environment variable GOOGLE_CLOUD_PROJECT is not set"
     exit 1
-else
-    echo -e "GOOGLE_CLOUD_PROJECT has been set to \e[33m${GOOGLE_CLOUD_PROJECT}\e[0m"
 fi
 
-# Confirm with the user that GOOGLE_CLOUD_PROJECT is set correctly
+echo -e "GOOGLE_CLOUD_PROJECT has been set to \e[33m${GOOGLE_CLOUD_PROJECT}\e[0m"
+
+# Prompt the user that GOOGLE_CLOUD_PROJECT has been set correctly, user will reply with a yes or no. If no, exit script with error message.
 read -p "Is the environment variable GOOGLE_CLOUD_PROJECT set correctly? (y/N) " -n 1 -r
-echo    # Move to a new line
+echo    # (optional) move to a new line
 if [[ ! $REPLY =~ ^[Yy]$ ]]; then
-    echo "Environment variable GOOGLE_CLOUD_PROJECT is not set correctly."
+    echo "Environment variable GOOGLE_CLOUD_PROJECT is not set correctly"
     exit 1
 fi
 
-declare -a dnsNames=("alpha.phac.gc.ca" "alpha.aspc.gc.ca" "alpha.phac-aspc.gc.ca" 
-                     "beta.phac.gc.ca" "beta.aspc.gc.ca" "beta.phac-aspc.gc.ca" 
-                     "data.phac.gc.ca" "donnes.aspc.gc.ca" "data-donnes.phac-aspc.gc.ca" 
-                     "api.phac.gc.ca" "ipa.aspc.gc.ca" "api-ipa.phac-aspc.gc.ca" 
-                     "open.phac.gc.ca" "ouvert.aspc.gc.ca" "open-ouvert.phac-aspc.gc.ca")
+# Define the zones to be created in an array with a colon as the delimiter
+declare -a zones=(
+    "alpha-phac-gc-ca:alpha.phac.gc.ca."
+    "alpha-aspc-gc-ca:alpha.aspc.gc.ca."
+    "alpha-phac-aspc-gc-ca:alpha.phac-aspc.gc.ca."
+    "beta-phac-gc-ca:beta.phac.gc.ca."
+    "beta-aspc-gc-ca:beta.aspc.gc.ca."
+    "beta-phac-aspc-gc-ca:beta.phac-aspc.gc.ca."
+    "data-phac-gc-ca:data.phac.gc.ca."
+    "donnes-aspc-gc-ca:donnes.aspc.gc.ca."
+    "data-donnes-phac-aspc-gc-ca:data-donnes.phac-aspc.gc.ca."
+    "api-phac-gc-ca:api.phac.gc.ca."
+    "ipa-aspc-gc-ca:ipa.aspc.gc.ca."
+    "api-ipa-phac-aspc-gc-ca:api-ipa.phac-aspc.gc.ca."
+    "open-phac-gc-ca:open.phac.gc.ca."
+    "ouvert-aspc-gc-ca:ouvert.aspc.gc.ca."
+    "open-ouvert-phac-aspc-gc-ca:open-ouvert.phac-aspc.gc.ca."
+)
 
-for dnsName in "${dnsNames[@]}"; do
-    zoneName="${dnsName//./-}" # Replace dots with dashes for the zone name
-    gcloud dns --project "${GOOGLE_CLOUD_PROJECT}" managed-zones create "${zoneName}" \
-        --description="" --dns-name="${dnsName}." --visibility="public" \
-        --dnssec-state="on" --log-dns-queries
+# Loop through the array and create each managed zone
+for zone_info in "${zones[@]}"; do
```

Comment:
for zone_name in "${!zones[@]}"; do
    dns_name=${zones[$zone_name]}
    gcloud dns --project "${GOOGLE_CLOUD_PROJECT}" managed-zones create "$zone_name" --description="" --dns-name="$dns_name" --visibility="public" --dnssec-state="on" --log-dns-queries
done